src/psyclone/psyir/nodes/omp_critical_directive.py

```diff
@@ -0,0 +1,66 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2021-2025, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors A. B. G. Chalk, STFC Daresbury Lab
+# -----------------------------------------------------------------------------
+
+''' This module contains the implementation of the OpenMP Critical Directive
+node.'''
+
+from psyclone.psyir.nodes.omp_directives import OMPRegionDirective
+
+
+class OMPCriticalDirective(OMPRegionDirective):
+    '''
+    OpenMP directive to inform that the contained region must only be executed
+    by a single thread at any time.
+
+    PSyclone does not currently do any analysis about critical regions, and
+    their correct usage is up to the user. In the future, we may try to
+    improve this (after issues like #3238).
+    '''
+    def begin_string(self) -> str:
+        '''
+        :returns: the opening string statement of this directive.
+        '''
+        return "omp critical"
+
+    def end_string(self) -> str:
+        '''
+        :returns: the ending string statement of this directive.
+        '''
+        return "omp end critical"
+
+
+# For automatic API documentation generation
+__all__ = ["OMPCriticalDirective"]
```

Comment:
I know this was mentioned in Teams and I didn't speak up, but I disagree in splitting omp to new files. I believe these will end up being very shallow classes (with similarly shallow tests). And I think splitting them all will just create more complexity that they really represent.

I don't disagree that the size of omp_directives.py needs to be reduced, but I think that other logic currently there has to be moved to a better place, not the shallow clases. And if we start moving them to multiple files it will make the actual refactoring more difficult. So unless @arporter and you disagree, I think I would still put this class in the main omp file.